test/e2e/framework/clusterprovider/kwokprovider.go

```diff
@@ -162,6 +162,7 @@ func (p *kwokClusterProvider) NewCluster(ctx context.Context, name string) (*fed
 func (p *kwokClusterProvider) StopCluster(ctx context.Context, name string) {
 	if _, ok := p.clusters.Load(name); ok {
```

Comment:
Do we need `LoadAndDelete` here?

Reply:
I think both works, but the reason why I returned a `stopClusterFn` is because I was not sure how to extend the interface of `Framework`, specifically what type of argument should be used to identify the cluster to be stopped - name, uid, cluster object?

_Preface:  `StopCluster` is currently only a method of the `ClusterProvider` interface, which is not directly exposed to users of the framework._